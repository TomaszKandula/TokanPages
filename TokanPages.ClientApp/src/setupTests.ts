// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
