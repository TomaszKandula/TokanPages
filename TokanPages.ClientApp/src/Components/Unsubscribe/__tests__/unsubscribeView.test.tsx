import "../../../setupTests";
import React from "react";
import { shallow } from "enzyme";
import UnsubscribeView from "../unsubscribeView";

describe("Test component: unsubscribeView.", () => 
{
    it("Renders correctly '<UnsubscribeView />' when content is loaded.", () => 
    {
        const tree = shallow(<UnsubscribeView bind=
        {{
            isLoading: false,
            caption: "Cancel your subscribtion",
            text1: "We are sorry to see you go...",
            text2: "...but we understand there are circumstances",
            text3: "Please contact us should you have any questions",
            buttonHandler: jest.fn(),
            buttonState: false,
            progress: false,
            buttonText: "Unsubscribe Now"
        }}/>);
        expect(tree).toMatchSnapshot();
    });
});
