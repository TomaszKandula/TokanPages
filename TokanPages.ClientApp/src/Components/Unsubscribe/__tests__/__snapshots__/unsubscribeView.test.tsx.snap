// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component: unsubscribeView. Renders correctly '<UnsubscribeView />' when content is loaded. 1`] = `
<section
  className="makeStyles-section-1"
>
  <WithStyles(ForwardRef(Container))
    maxWidth="sm"
  >
    <Styled(MuiBox)
      py={15}
    >
      <WithStyles(ForwardRef(Card))
        elevation={4}
      >
        <WithStyles(ForwardRef(CardContent))
          className="makeStyles-card-2"
        >
          <Styled(MuiBox)
            mb={3}
            textAlign="center"
          >
            <Styled(MuiBox)
              mb={2}
              mt={2}
            >
              <WithStyles(ForwardRef(Typography))
                component="h4"
                gutterBottom={true}
                variant="h4"
              >
                Cancel your subscribtion
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
            <Styled(MuiBox)
              mb={2}
              mt={5}
            >
              <WithStyles(ForwardRef(Typography))
                color="textSecondary"
                component="h6"
                variant="h6"
              >
                We are sorry to see you go...
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
            <Styled(MuiBox)
              mb={2}
              mt={5}
            >
              <WithStyles(ForwardRef(Typography))
                color="textSecondary"
                variant="body1"
              >
                ...but we understand there are circumstances
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
            <Styled(MuiBox)
              mb={7}
              mt={5}
            >
              <WithStyles(ForwardRef(Typography))
                color="textSecondary"
                variant="body1"
              >
                Please contact us should you have any questions
              </WithStyles(ForwardRef(Typography))>
            </Styled(MuiBox)>
            <WithStyles(ForwardRef(Button))
              color="primary"
              disabled={true}
              fullWidth={true}
              onClick={[MockFunction]}
              type="submit"
              variant="contained"
            >
              Unsubscribe Now
            </WithStyles(ForwardRef(Button))>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(CardContent))>
      </WithStyles(ForwardRef(Card))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Container))>
</section>
`;
